### DISCUSSION TITLE
Enable emoji reaction which connotes indifference

### DISCUSSION THREAD

**Discussion by tymokvo** [community/community#27905] [Other Feature Feedback, Questions, & Ideas]
Hello Hubbers! :smile_cat: 

In [this blog post](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/), it was announced that a subset of emoji would be supported for reactions. The reactions seem to fall into one of three "connotation" categories:

|Connotation|Emoji|
|---|---|
|Approval|:+1: :rocket: :smile: :heart: :tada:|
|Disapproval|:-1: :confused: |
|I'm aware of this|:eyes:|

I have encountered the situation on several projects where I'd like others to be able to express a connotation along the lines of "I'm indifferent to this change". For example, I have an open issue about reorganizing a monorepo of several related libraries into a single library, which will affect how the whole team works but is an operational change with minor functional changes. One team member has given positive feedback, but I'd like the others to have a way to express "I'm fine either way" without having to ping them to write a comment if they aren't interested.

On past teams, I have been in a similar situation where I would like to get the feedback of some subset of collaborators, but they are so focused on another part of the project that they don't really have an opinion on the proposal.

I propose adding another emoji reaction to the set: :shrug: to be able to do this so that it is easier to poll collaborators for this kind of response.

### Use one of the "Approval" emoji

I don't think this is always appropriate, as it seems that many people use this to indicate strong support of an issue (see [this discussion](https://github.com/orgs/community/discussions/5467) about :+1: indicating an "upvote") rather than "I have no opinion".

### Use :eyes: 
I don't think this is appropriate as people tend to use it to mean "I've seen this and I haven't had time to respond yet" which is different because the eventual response could be strongly positive or negative.

### Use :confused: 
:confused: is arguably the most ambiguous response and connotes "I don't get it" or "What's going on?". I think this is different than a situation where someone understands the tradeoffs, but simply can't think of strong advantages or disadvantages.

## Advantages
The main advantage that I see here is for collaborators to remove cognitive load of someone asking for feedback on a proposal _without_ having to come up with a strong positive or negative response if they have other priorities.

## Disadvantages
The main disadvantage that I see is that collaborators may not get a clear "go | no-go" response to a proposal if it's possible to express indifference. 

I think this can be ameliorated in cases where there needs to be a clear "go | no-go" by being explicit in a proposal in asking "please respond with :+1: / :-1: ...". And, as in the link above, there is already a strong tendency to associate "upvote/downvote" with this pattern.

**Comment by MrSarno** [comment]
> ### Use 👀
> I don't think this is appropriate as people tend to use it to mean "I've seen this and I haven't had time to respond yet" which is different because the eventual response could be strongly positive or negative.

This is weird to me.

If there has been no comment or emoji reaction from a member of my team, I would assume that they have not yet seen $theThing and wait for their input. You seem to value the distinction between that, and them clarifying that they have seen it, but are not yet ready to weigh in on the subject. I’m not really seeing the benefit of such a distinction. Could you clarify how you find it helpful?

**Reply by tymokvo** [reply]
Sure. At the simplest: no response is ambiguous, where an :eyes: response could indicate "I have eyes on this, but no positive/negative/neutral response."

Our team (like many) is distributed across many timezones and someone whose feedback is specifically needed may be aware of an issue, but was alerted to it at a time when they cannot respond for several hours (say, right before they leave for the day). This happens to me somewhat often and, personally, I use :eyes: to mean "I know about this and will get back to you" so that the author isn't left in an ambiguous state of "Is @tymokvo aware? Is it appropriate to DM/tag/etc. to get attention?".

**Reply by MrSarno** [reply]
Appreciate the response. 😃

I updooted you as I can’t find any obvious negative repercussions from this being done, and I have to imagine the work needed to implement it (and, thus, the opportunity cost) are minimal.

I still struggle to imagine a case in which the presence of the emoji would influence me; one of the limitations of reactions being you aren’t offered a time stamp, so it’s hard to know if the 👀 appeared an hour ago & is “fresh”, or if it was added within minutes of the issue’s creation by someone who has since forgotten. I feel like either there’s time pressure (ping), or there isn’t (don’t).

But, hey. Some groups are formal teams in companies and others are relative strangers in different time zones. Just because I don’t think this would help me now doesn’t mean it wouldn’t have value for others - or perhaps be useful to me in future.

**Reply by tymokvo** [reply]
Thanks for the input! My thinking with this is that, if users are generally amenable, it seems like it should be a one-line change for Github rather than a much bigger feature proposal (as came out in the discussion with @airtower-luna ).

**Comment by airtower-luna** [comment]
It's interesting to me that you interpret some of the reactions differently than I would, especially the :eyes: one. It'd use it for something along the lines of "this is fascinating/peculiar/suspicious". :sweat_smile: 

I guess my main conclusion from that is that if you need a clear statement better stick to words, except maybe if you have a clear definition in your team.

**Reply by tymokvo** [reply]
Interesting reaction interpretation! If I found a proposal or decision specifically peculiar/suspicious, I would probably call it out explicitly or use :confused: .

My main issue is that there's no way to express neutrality in any way. And asking collaborators "would you please take the time to write a comment if you're indifferent" seems like a lot of overhead in the first place to write the comment, and in the second place to have to read through all of the comments to come up with the poll of "who has definitely seen this and affirmed that they are neutral on the subject?"

I suppose another solution that I didn't mention would be an issue template or otherwise explicit text in the issue to the effect of "If you respond with :eyes: in this case, I will interpret it as you being indifferent". Maybe every project should also have an "Emoji Connotation Table" that is part of the contributor guidelines :wink: . In all seriousness, maybe a better feature proposal that would solve this issue in a more general way would be a repo configuration where teams can specify which emoji are available for reactions and what they indicate in the context of that repo. Similar to how tags can be configured and given a description.

**Reply by airtower-luna** [reply]
> In all seriousness, maybe a better feature proposal that would solve this issue in a more general way would be a repo configuration where teams can specify which emoji are available for reactions and what they indicate in the context of that repo.

That makes sense, though you might still run into trouble with people who don't read descriptions. But I guess it depends on your team how important that is. In a popular, public repository it probably is, with a well-connected team maybe not so much. :grin: 

I'm also wondering: Discussions [support polls](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#creating-a-poll), maybe polls for issues would make sense? I guess in that case they should allow changing your vote, e.g. as discussion progresses (maybe someone was convinced, or a proposal altered to their liking).

**Reply by tymokvo** [reply]
Hm, polls look promising for some use cases. Thank you for the suggestion! 

> ...with a well-connected team maybe not so much

I would disagree. Even a well-connected team has members join/leave, go on parental leave or vacation, etc. I think the UX of tags is nice in that the descriptions are so brief there's not a great reason to forego reading them. And, if the emoji configuration were an accepted feature proposal, the tooltip could easily be modified to render the description, similar to the way it works for tags.

It does seem like a fairly heavy solution to have to make two numbered objects (e.g. the initial poll and then turn it into an issue. And then close that with a PR). And, I agree, not being able to change your vote after discussion seems somewhat antithetical to the "Discussions" feature. But I will try it the next time this issue arises and see how it works.

**Reply by airtower-luna** [reply]
> Even a well-connected team has members join/leave, go on parental leave or vacation, etc. I think the UX of tags is nice in that the descriptions are so brief there's not a great reason to forego reading them.

I meant that with a well-connected team you're less likely to encounter issues with people not reading descriptions. And if someone still doesn't, they'll most likely be politely but firmly pointed at those descriptions. With the general public you'll always have people who ignore even the best documentation. :wink:
